### PR TITLE
Initial work for replacing suds with zeep

### PR DESCRIPTION
**Pull request by Jyrno42**
Adds tests for the following methods and verifies that they work

- StartSession
- MobileAuthenticate
- GetMobileAuthenticateStatus

For other parts see the umbrella issue #10

### CODE REVIEW

File: esteid/digidocservice/service.py [hunk 1]
```diff
@@ -1,37 +1,40 @@
...
     """ Unknown errors
     """
 
-    def __init__(self, command, params, *args, **kwargs):
+    def __init__(self, command, params, *args):
         self.command = command
         self.params = params
 
-        super().__init__(*args, **kwargs)
+        super(DigiDocException, self).__init__(*args)
```
Comment (not-much-io) [inline esteid/digidocservice/service.py:26]: Any reason for prefering super(ClassName, self) over super() ?

File: esteid/generic.py [hunk 2]
```diff
@@ -7,38 +7,56 @@
...
         """ This method returns the active digidoc session
             associated with the active request.
 
-            DigiDocService session is stored in request.session['__ddoc_session']
+            DigiDocService session is stored in request.session[I{DIGIDOC_SESSION_KEY}]
         """
 
         try:
-            return self.request.session['__ddoc_session']
+            return self.request.session[self.DIGIDOC_SESSION_KEY]
```
Comment (not-much-io) [inline esteid/generic.py]: return self.request.get(self.DIGIDOC_SESSION_KEY) ?

File: esteid/generic.py [hunk 3]
```diff
@@ -49,15 +67,20 @@ def destroy_digidoc_session(self):
...
 
-            del self.request.session['__ddoc_session']
+            del self.request.session[self.DIGIDOC_SESSION_KEY]
 
         except KeyError:
             pass
 
+    def destroy_digidoc_session_data(self):
+        # clear it just in case and then delete
+        self.request.session[self.DIGIDOC_SESSION_KEY] = {}
```
Comment (not-much-io) [inline esteid/generic.py]: Might be paranoid, but if some part of the code has retrieved the key dict and is holding on to it, then reassigning or deleting will leave the key there.

`self.request.session[self.DIGIDOC_SESSION_KEY].clear()` would clear the data and would propagate to every reference

**Review by not-much-io — CHANGES_REQUESTED** [review]
Some minor comments, otherwise looks good.

File: esteid/digidocservice/service.py (hunk 1 above)
Comment (Jyrno42) [reply esteid/digidocservice/service.py:26]: Py2 support

File: esteid/generic.py (hunk 3 above)
Comment (Jyrno42) [reply esteid/generic.py]: nice catch :+1:, did not think about that angle

File: requirements.txt [hunk 4]
```diff
@@ -1,5 +1,5 @@
 django>=1.5.1,<1.9
-lxml==3.5.0b1
-wheel==0.24.0
+lxml>=3.4,<4
+zeep>=2.4.0
 
-suds-jurko==0.6
+wheel==0.24.0
```
Comment (Jyrno42) [inline requirements.txt]: Update to `wheel==0.30.0`

File: esteid/digidocservice/service.py (hunk 1 above)
Comment (Jyrno42) [reply esteid/digidocservice/service.py:26]: Test output on python 2.7.12 without these changes:

```
py.test
======================================================================================================================== test session starts ========================================================================================================================
platform linux2 -- Python 2.7.12, pytest-3.0.4, py-1.4.34, pluggy-0.4.0
django settings: test_settings (from ini file)
rootdir: /home/jyrno42/Projektid/Progremine/thorgate/django-esteid, inifile: pytest.ini
plugins: django-2.9.1, cov-2.4.0
collected 9 items 

tests/tests.py ...F.....

============================================================================================================================= FAILURES ==============================================================================================================================
_______________________________________________________________________________________________________ TestMobileAuthenticate.test_mobileauthenticate_error ________________________________________________________________________________________________________

self = <tests.tests.TestMobileAuthenticate testMethod=test_mobileauthenticate_error>

    def test_mobileauthenticate_error(self):
        # From https://www.id.ee/?id=36381
        failing_numbers = [
            ('14212128027', 'EE', '+37200009', 302),
            ('38002240211', 'EE', '+37200001', 303),
        ]
    
        for id_code, country, phone_nr, error_code in failing_numbers:
            service = get_digidoc_service()
    
            with pytest.raises(DigiDocError) as e:
>               service.mobile_authenticate(id_code=id_code, country=country, phone_nr=phone_nr)

tests/tests.py:119: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
esteid/digidocservice/service.py:163: in mobile_authenticate
    }, no_raise=True)
esteid/digidocservice/service.py:390: in __invoke
    "Server result [/%s - %s]: %s" % (command, error_code, known_fault))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = DigiDocError(), error_code = 302, known_fault = 'Sertifikaat ei kehti (OCSP: TAGASI V\xc3\x95ETUD).', args = ('Server result [/MobileAuthenticate - 302]: Sertifikaat ei kehti (OCSP: TAGASI V\xc3\x95ETUD).',)

    def __init__(self, error_code, known_fault, *args):
        self.error_code = error_code
        self.known_fault = known_fault
    
>       super().__init__(*args)
E       TypeError: super() takes at least 1 argument (0 given)

esteid/digidocservice/service.py:37: TypeError
================================================================================================================ 1 failed, 8 passed in 34.45 seconds ================================================================================================================
Makefile:28: recipe for target 'test' failed
make: *** [test] Error 1
ERROR: InvocationError: '/usr/bin/make test'
______________________________________________________________________________________________________________________________ summary ______________________________________________________________________________________________________________________________
ERROR:   py27-django18: commands failed

```

Comment (not-much-io) [reply esteid/digidocservice/service.py:26]: Yeah absolutely, didn't know we were supporting py2, got it.

**Review by not-much-io — APPROVED** [review]

**Pull request merged**